### PR TITLE
Fix reading shadow filter quality from project settings in compatibility

### LINKED ISSUE #83380: Compatibility Renderer Soft Shadows at locked at PCF5 at runtime
### Godot version

4.2 beta 1

### System information

Godot v4.2.beta (ee118e7ff) - Windows 10.0.22621 - GLES3 (Compatibility) - AMD Radeon RX 6600 (Advanced Micro Devices, Inc.; 31.0.21031.1005) - Intel(R) Core(TM) i3-10105F CPU @ 3.70GHz (8 Threads)

### Issue description

Soft Shadows change in the editor properly, however when running the project, the Soft Shadows are locked at PCF5 (Directional and Spot, Omni doesn't have PCF implemented yet.)

### Steps to reproduce

Examine soft shadows and compare between editor and runtime

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by chybby**
Fixes #83380.
Supersedes #83943.

### CODE REVIEW

File: drivers/gles3/rasterizer_scene_gles3.cpp [hunk 1]
```diff
@@ -3289,6 +3289,9 @@ RasterizerSceneGLES3::RasterizerSceneGLES3() {
 	// Quality settings.
 	use_physical_light_units = GLOBAL_GET("rendering/lights_and_shadows/use_physical_light_units");
 
+	positional_soft_shadow_filter_set_quality(RS::ShadowQuality(int(GLOBAL_GET("rendering/lights_and_shadows/positional_shadow/soft_shadow_filter_quality"))));
```
Comment (AThousandShips) [inline drivers/gles3/rasterizer_scene_gles3.cpp:3292]:
```suggestion
	positional_soft_shadow_filter_set_quality((RS::ShadowQuality)(int)GLOBAL_GET("rendering/lights_and_shadows/positional_shadow/soft_shadow_filter_quality"));
```
Same below, matches other syntax and is more readable

**Comment by AThousandShips** [comment]
Copied from previous PR, by Akien:
> For Clay when reviewing: might be worth double checking in RendererSceneRenderRD::init() whether there's any other setting supported in the GL Compat renderer that's not properly initialized.

File: drivers/gles3/rasterizer_scene_gles3.cpp (hunk 1 above)
Comment (chybby) [reply drivers/gles3/rasterizer_scene_gles3.cpp]: Agreed this reads better as a cast. FYI I copied these lines from servers\rendering\renderer_rd\renderer_scene_render_rd.cpp RendererSceneRenderRD::init

**Review by clayjohn — APPROVED** [review]
Looks good to me! Great work!

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**